upgrade-tests/src/test/java/io/zeebe/test/RollingUpdateTest.java

```diff
@@ -0,0 +1,127 @@
+            .collect(Collectors.joining(","));
+
+    network = Network.newNetwork();
+
+    containers =
+        Arrays.asList(
+            withCloseable(new ZeebeBrokerContainer(OLD_VERSION)),
+            withCloseable(new ZeebeBrokerContainer(OLD_VERSION)),
+            withCloseable(new ZeebeBrokerContainer(OLD_VERSION)));
+    getConfiguredClusterBroker(0, containers);
+
+    nodeZeroContainer = getConfiguredClusterBroker(0, containers);
+    getConfiguredClusterBroker(1, containers);
+    getConfiguredClusterBroker(2, containers);
+  }
+
+  @Test
+  public void shouldBeAbleToRestartContainerWithSameVersion() {
+    // given
+    final var index = 0;
+    final var sameVersion = OLD_VERSION;
+    Startables.deepStart(containers).join();
+    nodeZeroContainer.shutdownGracefully(Duration.ofSeconds(30));
+
+    // when
+    final var zeebeBrokerContainer = replaceBrokerContainer(index, sameVersion);
+
+    // then
+    zeebeBrokerContainer.start();
+  }
+
+  @Test
+  public void shouldBeAbleToRestartContainerWithNewVersion() {
+    // given
+    final var index = 0;
+    final var newVersion = CURRENT_VERSION;
+    Startables.deepStart(containers).join();
+    nodeZeroContainer.shutdownGracefully(Duration.ofSeconds(30));
+
+    // when
+    final var zeebeBrokerContainer = replaceBrokerContainer(index, newVersion);
+
+    // then
+    zeebeBrokerContainer.start();
+  }
+
+  private ZeebeBrokerContainer replaceBrokerContainer(final int index, final String newVersion) {
+    final var broker = new ZeebeBrokerContainer(newVersion);
+    containers.set(index, broker);
+    nodeZeroContainer = getConfiguredClusterBroker(index, containers);
+    return broker;
+  }
+
+  private ZeebeBrokerContainer getConfiguredClusterBroker(
+      final int index, final List<ZeebeBrokerContainer> brokers) {
+    final int clusterSize = brokers.size();
+    final var broker = brokers.get(index);
+    final var hostName = "broker-" + index;
+    broker.withNetworkAliases(hostName);
+
+    return broker
+        .withNetwork(network)
+        .withEnv("ZEEBE_BROKER_NETWORK_HOST", "0.0.0.0")
+        .withEnv("ZEEBE_BROKER_NETWORK_ADVERTISED_HOST", hostName)
+        .withEnv("ZEEBE_BROKER_CLUSTER_CLUSTERNAME", "zeebe-cluster")
+        .withEnv("ZEEBE_BROKER_DATA_SNAPSHOTPERIOD", "1m")
+        .withEnv("ZEEBE_BROKER_DATA_LOGSEGMENTSIZE", "1MB")
+        .withEnv("ZEEBE_BROKER_NETWORK_MAXMESSAGESIZE", "1MB")
+        .withEnv("ZEEBE_BROKER_CLUSTER_NODEID", String.valueOf(index))
+        .withEnv("ZEEBE_BROKER_CLUSTER_CLUSTERSIZE", String.valueOf(clusterSize))
+        .withEnv("ZEEBE_BROKER_CLUSTER_REPLICATIONFACTOR", String.valueOf(clusterSize))
+        .withEnv("ZEEBE_BROKER_CLUSTER_INITIALCONTACTPOINTS", initialContactPoints)
+        .withLogConsumer(new Slf4jLogConsumer(LOG))
+        .withLogLevel(Level.DEBUG)
+        .withDebug(false);
+  }
+
+  private <T extends AutoCloseable> T withCloseable(final T closeable) {
```

Comment:
```suggestion
  private <T extends AutoCloseable> T asManagedCloseable(final T closeable) {
```

Reply:
this method was copied btw from your branch :D

Reply:
Everybody makes mistakes :sweat_smile: